utam-compiler/src/main/java/utam/compiler/helpers/TypeUtilities.java

```diff
@@ -160,6 +179,73 @@ public Class getClassType() {
     }
   }
 
+  public static class Element implements TypeProvider {
+    private final String name;
+    private String containingType = "";
+    private List<BasicElementInterface> basicInterfaces = new ArrayList<>();
+
+    Element(String name, String[] interfaceTypes) {
+      this(name, interfaceTypes, "");
+    }
+
+    Element(String name, String[] interfaceTypes, String containingType) {
+      this.name = name.substring(0, 1).toUpperCase() + name.substring(1) + "Element";
+      this.containingType = containingType;
+      for(String interfaceType : interfaceTypes) {
+        if (BasicElementInterface.isBasicType(interfaceType)) {
+          basicInterfaces.add(BasicElementInterface.asBasicType(interfaceType));
+        }
+      }
+    }
+
+    public static boolean isBasicType(String[] interfaceTypes) {
+      if (interfaceTypes == null) {
+        return true;
+      }
+      for (String interfaceType : interfaceTypes) {
+        if (!BasicElementInterface.isBasicType(interfaceType)) {
+          return false;
+        }
+      }
+      return true;
+    }
+
+    public static Element asBasicType(String name, String[] interfaceTypes) {
+      return asBasicType(name, interfaceTypes, "");
+    }
+
+    public static Element asBasicType(String name, String[] interfaceTypes, String containingType) {
+      if (interfaceTypes == null || interfaceTypes.length == 0) {
+        return new Element(name, new String[] { "actionable" }, containingType);
```

Comment:
let's not hardcode "actionable", we could use enum value and its name

Reply:
Perfectly reasonable. Fixed in next commit.